network-config-analyzer/K8sPolicyYamlParser.py

```diff
@@ -307,17 +320,19 @@ def parse_policy(self):
         if not res_policy.affects_egress and policy_spec.get('egress') is not None:
             self.syntax_error('A NetworkPolicy with egress field but no "Egress" in its policyTypes', policy_spec)
 
-        pod_selector = policy_spec.get('podSelector', policy_spec.get('pod_selector'))
-        if pod_selector is None:
-            self.syntax_error('A NetworkPolicy with no podSelector specified in its spec', policy_spec)
+        pod_selector = policy_spec.get('podSelector')
         res_policy.selected_peers = self.parse_label_selector(pod_selector)
         res_policy.selected_peers &= self.peer_container.get_namespace_pods(self.namespace)
 
-        for ingress_rule in policy_spec.get('ingress', []):
-            res_policy.add_ingress_rule(self.parse_ingress_rule(ingress_rule, res_policy.selected_peers))
+        ingress_rules = policy_spec.get('ingress', [])
+        if ingress_rules:
+            for ingress_rule in ingress_rules:
+                res_policy.add_ingress_rule(self.parse_ingress_rule(ingress_rule, res_policy.selected_peers))
 
-        for egress_rule in policy_spec.get('egress', []):
-            res_policy.add_egress_rule(self.parse_egress_rule(egress_rule))
+        egress_rules = policy_spec.get('egress', [])
+        if egress_rules:
+            for egress_rule in egress_rules:
+                res_policy.add_egress_rule(self.parse_egress_rule(egress_rule))
```

Comment:
can lines 328 and 333 can be omitted?
if no ingress/egress rules are specified then anyhow policy_spec.get('ingress', []) and policy_spec.get('egress', []) will return null and the code inside the following for loops to add the rule will not be executed.

Reply:
No, then we will get error message when having the following line in yaml file (copying from the test file that caused to add these lines):
 #3 empty (spare)
  egress:

if the lines (328, 333) are omitted
python reads this "value" as None , and None != [] , so it will try to iterate over it and print this error
TypeError: 'NoneType' object is not iterable
however k8s passes with no errors in this case
so I had to add the if statements in order to follow k8s behavior

Reply:
very good thanks for clarifying the reason